### ISSUE TITLE
Add support for DSL.execute() in Oracle

### ISSUE THREAD

**Issue by mnisius** [jOOQ/jOOQ#9885]
Hi. 

I tried to execute the following with jOOQ on Oracle 18

```java
   create.begin(
      execute(DSL.query("Select user from dual")),
      execute("Select user from dual")
    ).execute();
```

### Expected behavior and actual behavior:
Expected: The Statments are executed 
Actual: 
```
 org.springframework.jdbc.BadSqlGrammarException: jOOQ; bad SQL grammar [begin execute ?; end;]; nested exception is java.sql.SQLException: ORA-06550: Zeile 1, Spalte 15:
PLS-00103: Fand das Symbol "" als eines der folgenden erwartet wurde:

   := . ( @ % ; immediate
Das Symbol ":=" ersetzte "", um fortzufahren.

	at org.jooq_3.13.1.ORACLE18C.debug(Unknown Source)
	at org.springframework.jdbc.support.SQLErrorCodeSQLExceptionTranslator.doTranslate(SQLErrorCodeSQLExceptionTranslator.java:235)
	at org.springframework.jdbc.support.AbstractFallbackSQLExceptionTranslator.translate(AbstractFallbackSQLExceptionTranslator.java:72)
	at org.springframework.boot.autoconfigure.jooq.JooqExceptionTranslator.translate(JooqExceptionTranslator.java:92)
	at org.springframework.boot.autoconfigure.jooq.JooqExceptionTranslator.handle(JooqExceptionTranslator.java:81)
	at org.springframework.boot.autoconfigure.jooq.JooqExceptionTranslator.exception(JooqExceptionTranslator.java:55)
	at org.jooq.impl.ExecuteListeners.exception(ExecuteListeners.java:274)
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:386)
	at de.cosmosdirekt.vertrieb.batch.BatchApp.run(BatchApp.java:44)
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:784)
	... 10 common frames omitted
Caused by: java.sql.SQLException: ORA-06550: Zeile 1, Spalte 15:
PLS-00103: Fand das Symbol "" als eines der folgenden erwartet wurde:
``` 

### Steps to reproduce the problem
Use DSL.ececute in a Begin block and execute it

### Versions:
- jOOQ: 3.13.1
- Java: 11 
- Database Oracle 18
- OS: Windows
- JDBC Driver  ojdbc8

**Comment by lukaseder** [comment]
Thank you very much for your report.

As of jOOQ 3.13, `DSL.execute()` is only supported for the PostgreSQL dialect, not for Oracle yet. We should definitely add support for Oracle as well, of course, to produce an `EXECUTE IMMEDIATE` statement.

**Comment by mnisius** [comment]
Oh I was not aware that this feature is currently not supported for the Oracle dialect. If I had known I wouldn't reported it as a bug. 

The official manual is little bit misleading because the  Oracle dialect is explicit mentioned there. https://www.jooq.org/doc/3.13/manual-single-page/#procedural-execute

Thanks for adding it in the future tho.

**Comment by lukaseder** [comment]
> Oh I was not aware that this feature is currently not supported for the Oracle dialect. If I had known I wouldn't reported it as a bug.

No worries. As the documentation shows, this should have been included. It seems that the documentation was written before the implementation was finished. We'll definitely include this important statement soon, also for other dialects.

**Comment by lukaseder** [comment]
Implemented in jOOQ 3.14.0. Backport scheduled for 3.13.2 (#9944)
